ovos_phal_plugin_hotkeys/__init__.py

```diff
@@ -11,20 +11,142 @@
 # limitations under the License.
 #
 import json
+from json.decoder import JSONDecodeError
+from os import walk
+from os.path import exists, join, basename, dirname
+
 from ovos_bus_client.message import Message
 from ovos_plugin_manager.phal import PHALPlugin
 from ovos_utils.log import LOG
+from ovos_utils.xdg_utils import XDG_DATA_DIRS, XDG_DATA_HOME
+
+# TODO: Update ovos-PHAL to differentiate sj201 v6/v10
+# Currently, it only detects v6
+from ovos_PHAL.detection import is_respeaker_2mic, is_mycroft_sj201
+
+from ovos_plugin_manager.templates.phal import PHALValidator
 
 import ovos_phal_plugin_hotkeys.keyboard as keyboard
 
 
+def get_preconfigured_devices():
+    local_config = join(dirname(__file__), "config")
+    device_dir = "OpenVoiceOS/hotkey_devices"
+    devices = []
+    for (p, d, f) in walk(local_config):
+        for filename in f:
+            devices.append(join(p, filename))
+    for directory in XDG_DATA_DIRS:
+        for (p, d, f) in walk(join(directory, device_dir)):
```

Comment:
do we need a check `if os.path.isdir(join(directory, device_dir))` before walking it?

code style nitpick: paths are more readable with f strings instead of join

 `f"{dirname(__file__)}/config"` 
 `f"{directory}/OpenVoiceOS/hotkey_devices"`